core/rfb.js

```diff
@@ -942,24 +940,16 @@ export default class RFB extends EventTargetMixin {
             reason = this._sock.rQshiftStr(strlen);
         }
 
-        if (reason !== '') {
-
-            let event = new CustomEvent(
-                'securityfailure',
-                { detail: { status: security_result_status, reason: reason } });
-            this.dispatchEvent(event);
-
-            return this._fail('Security negotiation failed' + context +
-                              ' (reason: ' + reason + ')');
-        } else {
-
-            let event = new CustomEvent(
-                'securityfailure',
-                { detail: { status: security_result_status } });
-            this.dispatchEvent(event);
+        const event = new CustomEvent(
+            'securityfailure',
+            { detail: reason === ''
+                ? { status: security_result_status }
+                : { status: security_result_status, reason }
+            });
+        this.dispatchEvent(event);
```

Comment:
in other places you have removed the intermediate event variable in cases like these